spacy/default_config.cfg

```diff
@@ -80,6 +80,8 @@ eval_frequency = 200
 score_weights = {}
 # Names of pipeline components that shouldn't be updated during training
 frozen_components = []
+# Names of pipeline components that should set annotations on update
+set_annotations_on_update = []
```

Comment:
is "on update" clear for most users? Because the config hides some of the API calls, should we consider calling this something more intuitive like "during training"? Not that I want to make this any longer :|

Reply:
It's within the training block, so how about calling it `annotates`? It feels like that matches up with `frozen` better. If that's too terse, perhaps `set_annotations`, but that looks like it should be a boolean.

Reply:
or "overrides_gold"?

Reply:
`set_annotations_on_update` is intentional clunky and easy to rename.

I agree that `update` is unclear from the `[training]` block/script perspective. I know that `frozen_components` turns into `exclude`, but this feels weird. This needs to be the option for `nlp.update`, too, and it'd be nice to have the same for both?

Something with `predicted` would be better. Something along the lines of `annotate_predicted`?

Reply:
`persist_predictions`? (I realise I'm making things worse 😁 )

Reply:
`predictions` is bad, if we use it I want it to sound like `Example.predicted`.

Reply:
Planning to rename to `annotating_components` in `[training]` and `annotates` for `Language.update`.